rust-src/concordium_base/src/web3id/sdk/protocol.rs

```diff
@@ -0,0 +1,903 @@
+// TODO Use v1 presentation when ready.
+//! Types used in Concordium verifiable presentation protocol version 1.
+
+use crate::common::{cbor, Buffer, Deserial, Get, ParseResult, ReadBytesExt, Serial, Serialize};
+use crate::id::{id_proof_types::AtomicStatement, types::AttributeTag};
+use crate::web3id::{did, Web3IdAttribute};
+use crate::{hashes, id};
+use concordium_base_derive::{CborDeserialize, CborSerialize};
+use concordium_contracts_common::hashes::HashBytes;
+use sha2::Digest;
+use std::collections::HashMap;
+
+/// A verifiable presentation request that specifies what credentials and proofs
+/// are being requested from a credential holder.
+#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize, Serialize)]
+#[serde(tag = "type", rename = "ConcordiumVPRequestV1")]
+pub struct VerifiablePresentationRequest {
+    /// Presentation being requested.
+    #[serde(flatten)]
+    pub request: VerificationRequestData,
+    /// Blockchain transaction hash that anchors the request.
+    #[serde(rename = "requestTX")]
```

Comment:
I know it says "requestTX" in the ADR, but can we name it something a little nicer? We normally don't abbreviate "transaction" to "tx" in the API. I've gone with `transactionRef` in the web SDK.